fivox/cuda/simpleLFP.cu

```diff
@@ -0,0 +1,128 @@
+    const int zIndex = (( threadId / volInfo.dimensions.x )
+                        / volInfo.dimensions.y ) % volInfo.dimensions.z;
+
+    const float voxelPosX = xIndex * volInfo.voxelSize + volInfo.origin.x;
+    const float voxelPosY = yIndex * volInfo.voxelSize + volInfo.origin.y;
+    const float voxelPosZ = zIndex * volInfo.voxelSize + volInfo.origin.z;
+
+    // Compute directly the inverted value to gain performance in the for loop
+    const float cutOffDistance = 1.f / params.cutoff;
+
+    float current( 0.f );
+    for( unsigned int i = 0; i < params.numEvents; ++i )
+    {
+        const float value( values[i] );
+
+        const float distanceX = voxelPosX - eventsX[i];
+        const float distanceY = voxelPosY - eventsY[i];
+        const float distanceZ = voxelPosZ - eventsZ[i];
+
+        const float distance2( distanceX * distanceX +
+                               distanceY * distanceY +
+                               distanceZ * distanceZ );
+
+        // Use the reciprocal of sqrt (faster in AVX assembly)
+        const float length = 1.f / sqrtf( distance2 );
+        // Comparison is inverted, as we are using the reciprocal values
+        if( length < cutOffDistance )
+            continue;
+
+        // If center of the voxel within the event radius, use the
+        // voltage at the surface of the compartment (at 'radius' distance)
+        const float radius( radii[i] );
+        // Comparison is inverted, as we are using the reciprocal values
+        // (radius is already stored like that from the loader)
+        if( length > radius )
+            current += value * radius; // mA
+        else
+            current += value * length; // mA
+    }
+    // voltageFactor =  1 / (4 * PI * conductivity),
+    // with conductivity = 1 / 3.54 (siemens per meter)
+    const float voltageFactor = 0.281704249f;
+    // output voltage in mV
+    output[threadId] = voltageFactor * current;
+}
+
+float simpleLFP( float* posX, float* posY, float* posZ, float* radii,
+                 float* values, const Parameters& parameters,
+                 const VolumeInfo& volInfo, float* output )
+{
+    cuda::Parameters* cudaParameters;
+    cuda::VolumeInfo* cudaVolInfo;
+    gpuErrchk( cudaMalloc( (void**)&cudaParameters, sizeof( cuda::Parameters)));
+    gpuErrchk( cudaMalloc( (void**)&cudaVolInfo, sizeof( cuda::VolumeInfo )));
+
+    gpuErrchk( cudaMemcpy( cudaParameters, &parameters,
+                           sizeof( cuda::Parameters ), cudaMemcpyHostToDevice));
+    gpuErrchk( cudaMemcpy( cudaVolInfo, &volInfo, sizeof( cuda::VolumeInfo ),
+                           cudaMemcpyHostToDevice ));
+
+    const int numVoxels = volInfo.dimensions.x *
+                          volInfo.dimensions.y *
+                          volInfo.dimensions.z;
+
+    // 1D grid of 1D blocks
+    dim3 blockSize( 512, 1 );
+    dim3 gridSize( ( numVoxels + blockSize.x * blockSize.y - 1 ) /
+                   ( blockSize.x * blockSize.y ), 1 );
```

Comment:
I wanted to keep it generic, to be able to change the dimensions easily. We still need to experiment and compare with different sizes.